### PR TITLE
world.remove now queues bodies for deletion (updated)

### PR DESCRIPTION
**Pull request by cgmartin**
Reimplementation of #68 for a potential fix for #306.

Sharing this as a possible workaround to an exception thrown when calling `removeBody()` immediately upon receiving a body "collide" event:

```
cannon.js:14361 Uncaught TypeError: Cannot read property '_wakeUpAfterNarrowphase' of undefined
```

To replicate issue, in this PR branch, change the code in `demos/events.html` to:

``` js
// lines 61-64 ...
if(options.destroyOnCollision){
    demo.removeVisual(sphereBody);
    world.removeImmediate(sphereBody); // <-- change this to repro error
}
```

**Warning: This is a breaking API change.** 

Without the change in RaycastVehicle.js, the following unit test failure occurs:

```
Testing RaycastVehicle.js..........F
>> removeFromWorld
>> Error: false == true
>> at Object.removeFromWorld (test/RaycastVehicle.js:107:14)
>> at runCallback (timers.js:574:20)
>> at tryOnImmediate (timers.js:554:5)
>> at processImmediate [as _immediateCallback] (timers.js:533:5)
```

Based on the last conversation in #68 I'm not confident that this solution is desired. But if the maintainers would like to move forward with this PR, I could use help with unit tests that can reproduce the `Uncaught TypeError` scenario (removing a body on "collide" event).

### CODE REVIEW

**Comment by cgmartin** [comment]
An alternative non-breaking API change could be to keep `world.remove()/world.removeBody()` as it is currently with its immediate remove behavior, but add a new `world.queueRemoveBody()` method instead.

**Comment by schteppe** [comment]
Hi, I appreciate the PR, but do read this: https://github.com/schteppe/p2.js/issues/212

TL;DR to fix this problem, one would have to queue not only the body removals, but also the additions as well. And it doesn't stop there, constraints, shapes and everything else needs add/remove queues.
To avoid this queue hell, the solution in p2.js (and [box2d](http://www.iforce2d.net/b2dtut/removing-bodies)) is to just throw an error if a user tries to do this.
Please let me know what you think about this!

**Comment by cgmartin** [comment]
@schteppe makes sense, thanks for the quick response! Will use try/catch going forward. Closing this PR.

**Comment by cgmartin** [comment]
Also close #68 ?

**Comment by schteppe** [comment]
Yep, done! Thanks.

**Comment by Bab64** [comment]
Hi. Just a naive question: when using "removeBody" for body "b1", do we have to remove any constraint ( any other stuff as well) on "b1" just **before**? For instance, if I have a spring between "b1" and "b2", do I have to remove this spring before removing "b1" or **does Cannon do the job for me inside "removeBody"**? Thanks for replying!
